workerpool/pool.go

```diff
@@ -114,53 +149,39 @@ func newWorker(id uint, log *zap.SugaredLogger, errors chan<- workerError, close
 }
 
 func (w *worker) run() {
-	w.log.Debugf("kicked off worker #%02d...", w.id)
+	w.log.Debugw("kicked off worker", "workerID", w.id)
 	for {
 		select {
 		case <-w.done:
-			w.log.Debugf("trapped done signal for worker %d...", w.id)
+			w.log.Debugw("trapped done signal", "workerID", w.id)
 			return
-		case c := <-w.connection:
-			w.log.Debugf("worker %d started job:  %s", w.id, c.ID)
-			if err := w.handleConnection(c); err != nil {
-				w.errors <- workerError{id: w.id, connectionID: c.ID, err: err}
-				return
+
+		default:
+			// perform the actual connection here
+			for i := 0; i < 3; i++ {
```

Comment:
Why remove the function call? I'm not precisely sure how this is going to work until we implement the connection part but if this is indeed part of the source interface then I'd suspect we'd end up calling that function.

Let's leave this for now and can revisit when the connector part is ready.